packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts

```diff
@@ -24,35 +28,29 @@ export async function run(context: $TSContext) {
     return;
   }
 
-  const environmentName = args.environmentName ?? context.exeInfo.localEnvInfo.envName;
-  const appId = args.appId ?? context.exeInfo.teamProviderInfo[environmentName].awscloudformation.AmplifyAppId;
-  const amplifyUIBuilder = await getAmplifyUIBuilderService(context, environmentName, appId);
   const components = existingComponents.entities;
   if (!components.length) {
     printer.debug(`No UIBuilder components found in app ${appId} from env ${sourceEnvName}. Skipping component clone process.`);
     return;
   }
-  const promises = components.map(async (component: any) => {
-    return await amplifyUIBuilder
-      .createComponent({
-        appId,
-        environmentName,
-        componentToCreate: {
-          bindingProperties: component.bindingProperties,
-          children: component.children,
-          componentType: component.componentType,
-          name: component.name,
-          overrides: component.overrides,
-          properties: component.properties,
-          sourceId: component.sourceId,
-          variants: component.variants,
-        },
-      })
-      .promise();
-  });
+  const promises = components.map(component => studioClient.createComponent(
```

Comment:
We need to make sure we throttle this so we don't get 429'd

Reply:
We can await each creation and run that sequentially as opposed to promise.all or do you have preferred rate to limit batch requests?